tests/core/utilities/test_valid_transaction_params.py

```diff
@@ -38,19 +38,32 @@ def test_assert_valid_transaction_params_invalid_param():
 
 
 def test_extract_valid_transaction_params():
-    input = {
+    from copy import deepcopy
+    base_input = {
         'from': '0x0',
         'to': '0x0',
         'gas': 21000,
         'gasPrice': 5000000,
         'value': 1,
-        'data': '0x0',
         'nonce': 2,
         'chainId': 1,
     }
+
+    input = deepcopy(base_input)
+    input['data'] = '0x0'
     valid_transaction_params = extract_valid_transaction_params(input)
     assert valid_transaction_params == input
 
+    in_input = deepcopy(base_input)
+    in_input['input'] = '0x0'
+    in_input['data'] = '0x0'
+    valid_transaction_params = extract_valid_transaction_params(in_input)
+    assert valid_transaction_params == input
+
+    input['input'] = '0x1'
+    with pytest.raises(AttributeError, match=r'.* You need to resolve this conflict.'):
+        extract_valid_transaction_params(input)
+
```

Comment:
This looks like it should really be just like the old test, but with parametrized inputs and expected values. I think there are a couple more cases to cover, too. Will be easier to see in paramaterized inputs.

Reply:
the test envelope right now is:
1: data, no input :: data
2: data, input, data == input :: data
3: data, input,  data != input :: Error  (for that i need to change the value, e.g., '0x1')

to add:
4: no data, input :: data  (from input)
adding 5: no data, no input seems overkill but i'm happy to oblige.